trin-core/src/portalnet/mod.rs

```diff
@@ -15,6 +17,49 @@ mod overlay_service;
 
 pub type Enr = discv5::enr::Enr<CombinedKey>;
 
+construct_uint! {
```

Comment:
So, this was the home for `U256` - but it doesn't feel like the appropriate place for these uint definitions/tests. Personally, I don't think we should have any code inside our `mod.rs` files - it should be strictly for `pub mod ..` declarations. How do you feel about moving `U256` and `U512` definitions and tests over to `trin-core/src/types/uint.rs` or something similar?

Reply:
Yeah that makes sense to me since those are types. Should I address that in this PR?

Reply:
Probably not this pr since there's a lot going on already, but i'd just take a not to address it in a follow-up pr after this gets merged